mobile/test/trade_test.dart

```diff
@@ -0,0 +1,114 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get_10101/common/amount_denomination_change_notifier.dart';
+import 'package:get_10101/common/domain/model.dart';
+import 'package:get_10101/features/trade/domain/api_response.dart';
+import 'package:get_10101/features/trade/domain/response_status.dart';
+import 'package:get_10101/features/trade/order_change_notifier.dart';
+import 'package:get_10101/features/trade/submit_order_change_notifier.dart';
+import 'package:get_10101/features/trade/trade_screen.dart';
+import 'package:get_10101/features/trade/trade_theme.dart';
+import 'package:get_10101/features/trade/trade_value_change_notifier.dart';
+import 'package:get_10101/util/constants.dart';
+import 'package:go_router/go_router.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:provider/provider.dart';
+
+@GenerateNiceMocks([MockSpec<TradeValuesService>()])
+import 'package:get_10101/features/trade/application/trade_values_service.dart';
+
+@GenerateNiceMocks([MockSpec<OrderService>()])
+import 'package:get_10101/features/trade/application/order_service.dart';
+
+import 'trade_test.mocks.dart';
+
+final GoRouter _router = GoRouter(
+  initialLocation: TradeScreen.route,
+  routes: [
+    GoRoute(
+        path: TradeScreen.route,
+        builder: (BuildContext context, GoRouterState state) {
+          return const TradeScreen();
+        }),
+  ],
+);
+
+class TestWrapperWithTradeTheme extends StatelessWidget {
+  final Widget child;
+  const TestWrapperWithTradeTheme({super.key, required this.child});
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp.router(
+      // TODO: We could consider using the Navigator instead of GoRouter to close the bottom sheet again
+      // Need GoRouter otherwise closing the bottom sheet after confirmation fails
+      routerConfig: _router,
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+        extensions: const <ThemeExtension<dynamic>>[
+          // Need the trade theme otherwise the trade widgets that rely on it can't find it on the context and fail to render
+          TradeTheme(),
+        ],
+      ),
+    );
+  }
+}
+
+void main() {
+  testWidgets('Buy order flow until submission', (tester) async {
```

Comment:
Is this the approach to call tests like this in flatter?

I personally would like to have the naming simular to what we do in rust, e.g. `when user fills out order submission form, orderservice gets invoced with correct values.`

Reply:
That's a horrible name, I will amend it using the `given` `when` `then` pattern :)